org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/builder/State.java

```diff
@@ -62,7 +62,7 @@ public class State {
 // keyed by the project relative path of the type (i.e. "src1/p1/p2/A.java"), value is a ReferenceCollection or an AdditionalTypeCollection
 Map<String, ReferenceCollection> references;
 // keyed by qualified type name "p1/p2/A", value is the project relative path which defines this type "src1/p1/p2/A.java"
-public Map<String, String> typeLocators;
+public Map<String, Map<Integer,String>> typeLocators;
```

Comment:
Please update the comment defining the format of this map.

Wait a minute: this change would break org.eclipse.pde.api.tools!

Reply:
> Wait a minute: this change would break org.eclipse.pde.api.tools!

We should update API tools then to adapt to this change of (internal) API.

Reply:
Done